### PR TITLE
New version selection experience

### PR DESCRIPTION
**Pull request by tobias-tengler**
#### Changes

- New dropdown control to select version of current product
image
- Warning for outdated documentation
image
- Cleanup of navigation rendering logic

Preview: https://chilly-version-selection.netlify.app/docs/hotchocolate/

### CODE REVIEW

**Review by arif-hanif — DISMISSED** [review]

File: website/src/components/doc-page/doc-page.tsx [hunk 1]
```diff
@@ -117,6 +117,18 @@ export const DocPage: FunctionComponent<DocPageProperties> = ({
                     </Button>
                   </ResponsiveMenu>
                 </ResponsiveMenuWrapper>
+                {product.version !== "" ? (
+                  <OutdatedDocumentationWarning>
+                    <div>
```
Comment (rstaib) [inline website/src/components/doc-page/doc-page.tsx:122]: Should we do an one-liner?

Comment (rstaib) [reply website/src/components/doc-page/doc-page.tsx:122]: I think it eats up to much space :-)

**Review by rstaib — CHANGES_REQUESTED** [review]

**Comment by rstaib** [comment]
Looks pretty good!

**Comment by tobias-tengler** [comment]
@rstaib done!
Could we maybe first get #3879 in, then I can re-use the variables I declared there here.

**Comment by rstaib** [comment]
@tobias-tengler merged #3879 👍

**Review by rstaib — APPROVED** [review]
Looks cool!

**Pull request merged**